src/NServiceBus.Core/DelayedDelivery/TimeoutManager/DispatchTimeoutBehavior.cs

```diff
@@ -31,7 +31,8 @@ public DispatchTimeoutBehavior(IDispatchMessages dispatcher, IPersistTimeouts pe
 
             var outgoingMessage = new OutgoingMessage(context.MessageId, timeoutData.Headers, timeoutData.State);
             var transportOperation = new TransportOperation(outgoingMessage, new UnicastAddressTag(timeoutData.Destination), dispatchConsistency);
-            await dispatcher.Dispatch(new TransportOperations(transportOperation), context.Context).ConfigureAwait(false);
+            var transportTransaction = context.Context.GetOrCreate<TransportTransaction>();
```

Comment:
Ah, I think I see why this is they way it is here, since `MessageContext.TransportTransaction` get stored in the Extensions  context bag in the `TransportReceiveContext` 

Ok, using `GetOrCreate` makes sense now.